api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -115,6 +115,13 @@
    */
   UpdateSchema updateSchema();
 
+  /**
+   * Create a new {@link UpdateNameMapping} to update name mapping of this table and commit the change.
+   *
+   * @return a new {@link UpdateNameMapping}
+   */
+  UpdateNameMapping updateNameMapping();
```

Comment:
While I think it makes sense to update the mapping programmatically, I don't see much value in exposing it as part of the table API. We want to keep the Table API small, so if we can handle this by using a separate API that consumes and produces JSON, then that is preferred. Also, we may have more than one mapping in the future, so keeping this decoupled also helps out for those use cases.